e2e_tests/tests/experiment/test_core.py

```diff
@@ -467,8 +473,11 @@ def test_max_concurrent_trials(name: str, searcher_cfg: str) -> None:
         # the first two trials.
         exp.cancel_single(experiment_id)
 
+        trials = exp.wait_for_at_least_n_trials(experiment_id, 4)
+        assert len(trials) == 4, trials
+        time.sleep(3)
```

Comment:
i was trying so hard to avoid this. if we're going to do this, we prob can remove line 474

Reply:
Avoiding it to avoid a sleep?

Without the sleep, how can you actually know max_concurrent_trials is even working?

Really... without some synchronization signal, it's still only _usually_ going to fail when max_concurrent_trials isn't respected.